app/models/user.js

```diff
@@ -195,7 +157,14 @@ exports.addModel = function(database) {
       that.id = uuid.v4()
 
       that.validateOnCreate()
-        .then(function(user) { return user.updateHashedPassword() })
+        .then(function(user) {
+          if (_.isFunction(user.hashedPassword)) {
+            var pwd = user.hashedPassword();
+            return user.updatePassword(pwd, pwd);
+          } else {
+            resolve(user);
```

Comment:
YOU GOT ME!! // smashed his laptop to pieces. what a mess it was!

I will update wiki page, thank you! I was really under impression it's been written down already.